src/components/KTable/KTable.vue

```diff
@@ -843,11 +887,23 @@ watch(() => props.searchInput, (newValue) => {
   }
 }, { immediate: true })
 
-watch(() => [query.value, page.value, pageSize.value], ([newQuery, /* newPage */, /* newPageSize */, oldQuery]) => {
-  if (newQuery === '' && newQuery !== oldQuery) {
-    revalidate()
-  } else {
-    debouncedRevalidate()
+const isRevalidating = ref(false)
+watch(() => [query.value, page.value, pageSize.value], async (newData, oldData) => {
```

Comment:
suggestion: you don't need a getter when watching reactive refs
```suggestion
watch([query, page, pageSize], async (newData, oldData) => {
```